src/app/parser/impl/sol/SolParser.ts

```diff
@@ -0,0 +1,74 @@
+//
+// LESERKRITIKK v2 (aka Reader Critics)
+// Copyright (C) 2017 DB Medialab/Aller Media AS, Oslo, Norway
+// https://github.com/dbmedialab/reader-critics/
+//
+// This program is free software: you can redistribute it and/or modify it under
+// the terms of the GNU General Public License as published by the Free Software
+// Foundation, either version 3 of the License, or (at your option) any later
+// version.
+//
+// This program is distributed in the hope that it will be useful, but WITHOUT
+// ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+// FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License along with
+// this program. If not, see <http://www.gnu.org/licenses/>.
+//
+
+import * as Cheerio from 'cheerio';
+
+import ArticleAuthor from 'base/ArticleAuthor';
+
+import AbstractLabradorParser from 'app/parser/AbstractLabradorParser';
+import IteratingParserItem from 'app/parser/IteratingParserItem';
+import {cfEmailDecode} from 'app/parser/util/EmailDecode';
+
+export default class SolParser extends AbstractLabradorParser {
+
+	// Implement AbstractParser
+
+	protected parseVersion() : Promise <string> {
+		return Promise.resolve(
+			this.select('div.meta').find('meta[itemprop="dateModified"]').attr('content')
+		);
+	}
+
+	protected parseByline() : Promise <ArticleAuthor[]> {
+		const authorWrap = this.select('div.byline').find('span.person').toArray();
+		return Promise.resolve(authorWrap.map(wrap => {
+			const name = ''.concat(this.select(wrap).find('span.name').find('span.firstname').text(),
+				this.select(wrap).find('span.name').find('span.lastname').text());
+			const encodedMail = this.select(wrap).find('a[rel="author"]').attr('href');
+			let mail;
+
+			if (encodedMail.includes('mailto:')) {
+				mail = encodedMail.replace('mailto:', '');
+			} else if (encodedMail.includes('/cdn-cgi/l/email-protection#')) {
+				mail = cfEmailDecode(encodedMail.replace('/cdn-cgi/l/email-protection#', ''));
+			}
+
+			return {
+				name,
+				email: mail === undefined ? undefined : mail,
```

Comment:
Why do you need such construction?
If mail is undefined, it will be undefined

Reply:
the same was implemented in other parser by Phillipe, i'am trying to write test , after that maybe can say what he meant, but not sure

Reply:
There was ` mail.replace('mailto:', '')`, but you moved this thing up :)

Reply:
look, please,  this lines
if (encodedMail.includes('mailto:')) {
				mail = encodedMail.replace('mailto:', '');
			} else if (encodedMail.includes('/cdn-cgi/l/email-protection#')) {
				mail = cfEmailDecode(encodedMail.replace('/cdn-cgi/l/email-protection#', ''));
			}